pkg/controller/v1alpha2/applicationconfiguration/render.go

```diff
@@ -325,7 +325,8 @@ func renderWorkload(data []byte, p ...Parameter) (*unstructured.Unstructured, er
 	return &unstructured.Unstructured{Object: w.UnstructuredContent()}, nil
 }
 
-func renderTrait(data []byte, _ ...Parameter) (*unstructured.Unstructured, error) {
+// RenderTrait renders Trait to *unstructured.Unstructured format
+func RenderTrait(data []byte, _ ...Parameter) (*unstructured.Unstructured, error) {
 	// TODO(negz): Is there a better decoder to use here?
```

Comment:
we should create a common function for it instead of changing it public, and this common function could be in helper lib or anywhere else instead of the main logic lib.